tools/test

```diff
@@ -17,6 +17,14 @@ Options:
               with --full.
   --fix       Fix issues found, where possible.
   --full      Run on all files, not only changed files.
+  --platform [ios|android|both|sloppy]
+              Target iOS-specific or Android-specific code, or both.
+              "sloppy", the default, chooses one platform arbitrarily
+              for some test suites, so it doesn't give full coverage.
+              It exists just to decrease the running time for local
+              testing, given that much of our code isn't
+              platform-specific.
```

Comment:
> Maybe I'll just make a pass over it after merge.

Sure, that'd be helpful, thanks. 🙂

> where it can tell the difference

I'm not sure I follow the reasoning for choosing to run on both platforms vs. just one. Is it that in one case (Jest), we're likely to see the same successes and failures on both platforms?

Reply:
By "it" I meant tools/test (but didn't say that clearly).

So for the Android-native and future iOS-native builds and tests, we go ahead and run them in the default configuration (the one optimized for interactive use), because we know we can do so without duplicating anything.

Also in a future where we had some clever way of identifying which Jest tests actually depended meaningfully on the value of `Platform.OS` (or whatever else might vary between those two Jest "platform" settings), I'd want the default behavior to be to run those tests both ways, while continuing to run the other Jest tests only once. (And if we were confident that that clever way didn't miss anything -- maybe had false positives, but no false negatives -- then I'd want to do that even in CI.)

But as it stands, we know that the great majority of the Jest tests don't depend on the platform, and we can't in any automated way tell which are the exceptions. So by default we only run the tests once, and pick arbitrarily which platform setting to use.

Reply:
Sounds good. I've revised the description with this in mind, but I'm not sure I've found the best/most concise wording—please feel free to discard and start fresh if you like. 🙂